azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -890,4 +865,31 @@ public void afterExecute(Runnable r) {
     submittedFlows.remove(r);
   }
 
+  /**
+   * This shuts down the flow runner. The call is blocking and awaits execution of all jobs.
+   */
+  public void shutdown() {
+    logger.warn("Shutting down FlowRunnerManager...");
+    executorService.shutdown();
+    boolean result = false;
+    while (!result) {
+      logger.info("Awaiting Shutdown. # of executing flows: " + getNumRunningFlows());
+      try {
+        result = executorService.awaitTermination(1, TimeUnit.MINUTES);
+      } catch (InterruptedException e) {
+        logger.error(e);
+      }
+    }
+    logger.warn("Shutdown FlowRunnerManager complete.");
+  }
+
```

Comment:
I'm new to ExecutorService, but the logic in my mind should be like:
if(executorService.awaitTermination(1, TimeUnit.MINUTES)) 
   print "shut down completes"
else
   shutdownnow();